### DISCUSSION TITLE
How can I separate hms and gms services?

### DISCUSSION THREAD

**Discussion by orhanoksuzz** [mmazzarolo/react-native-universal-monorepo#36] [Q&A]
How do you separate Huawei and Google services by platform? Is there an exemplary approach to this?

**Comment by orhanoksuzz** [comment]
my example approach is dividing index.hms.js or index.gms.js at component level but I am getting error here. I couldn't ignore it at runtime level because the metro bundler caught both files. How can I parse index.gms.js and index.hms.js with metro bundler?

**Reply by mmazzarolo** [reply]
Hey @orhanoksuzz , I'm not familiar with the Huawei/Google platform support on React-Native... is the `hms.js`/`gms.js` officially supported by React Native/Metro?

**Reply by orhanoksuzz** [reply]
@mmazzarolo no it is not supported. In fact, the two platforms are separated on the basis of services with android infrastructure. For example, standard android devices use firebase and google services (push,map etc). Since Huawei devices do not support google, they use their own services. They are called Huawei mobile services. (push, map etc). There are package installations and settings for each. I separated android folder settings like different platform. Since the library usage is different in both services, I import them in different js files. In order to distinguish them, I divided them into index.gms.js and index.hms.js. ( gms => google mobile services) as hms => huawei mobile services). For now, when I build, I import which platform I want to import in index.js. If I'm going to get a build with example gms, I import index.js in the component as index.gms.js, that's how I get build. It will be much better if there is a formula to do this in metro config

**Comment by orhanoksuzz** [comment]
@mmazzarolo  Do you have an idea about the solution?

**Reply by mmazzarolo** [reply]
(sorry, I'm really busy latetly) 
Not really, if it's not supported by metro it's not in scope with the project :/

**Comment by orhanoksuzz** [comment]
@mmazzarolo 

Actually, I'm trying to do this for the solution.
Specifying a library name in the repo, for example @react-native-services, then configuring it to call @react-native-gms or @react-native-hms instead of @react-native-services when packaging. I think this is possible with craco, but I couldn't find how to do it in the native part.
